### PR TITLE
Fix invalid legacy permission fetching for comment mentions

### LINKED ISSUE #23970: Cannot Create Comment with @mention in Directus – “FORBIDDEN” Error
### Describe the Bug

When attempting to use the @mention feature in the comments section in Directus, we encounter a “FORBIDDEN” error message. This issue persists even after performing a fresh installation and testing with a newly created item.

JSON:
```
{
  "errors": [
    {
      "message": "You don't have permission to access fields \"admin_access\", \"app_access\" in collection \"directus_roles\" or they do not exist. Queried in \"role\".",
      "extensions": {
        "reason": "You don't have permission to access fields \"admin_access\", \"app_access\" in collection \"directus_roles\" or they do not exist. Queried in \"role\".",
        "code": "FORBIDDEN"
      }
    }
  ]
}
```

### To Reproduce

	1.	Set up a fresh installation of Directus (self-hosted Docker image).
	2.	Create a new item within any collection.
	3.	Attempt to comment on the item and use @mention to tag another user.
	4.	Observe the “FORBIDDEN” error in the response.

### Directus Version

v.11.1.2

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

_No response_

### PR DESCRIPTION
**Pull request by licitdev**
## Scope

What's changed:

- Fix invalid legacy permissions fetching for comments tagging

## Potential Risks / Drawbacks

- Lorem ipsum dolor sit amet
- Consectetur adipiscing elit

## Review Notes / Questions

- I would like to lorem ipsum
- Special attention should be paid to dolor sit amet

---

Fixes #23970

### CODE REVIEW

File: api/src/services/comments.ts [hunk 1]
```diff
@@ -147,18 +148,23 @@ export class CommentsService extends ItemsService {
...
 				role: user['role']?.id ?? null,
-				admin: user['role']?.admin_access ?? null,
-				app: user['role']?.app_access ?? null,
+				admin: false,
+				app: false,
 				roles: await fetchRolesTree(user['role']?.id, this.knex),
 				ip: null,
 			};
 
+			const userGlobalAccess = await fetchGlobalAccessForUser(accountability, this.knex);
```
Comment (licitdev) [inline api/src/services/comments.ts:163]: @hanneskuettner Should this be `fetchGlobalAccess` instead? 🤔 Would it work with public user?

Comment (hanneskuettner) [reply api/src/services/comments.ts:163]: You are correct, it should be `fetchGlobalAccess`, so the access from the users roles are considered as well.

Comment (licitdev) [reply api/src/services/comments.ts]: Thank you! Resolved via https://github.com/directus/directus/pull/23972/commits/7524de973c39f797da5ad7902e346089400023ad

File: .changeset/tough-snails-accept.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+Fixed invalid legacy permissions fetching for comments tagging
```
Comment (ComfortablyCoding) [inline .changeset/tough-snails-accept.md:5]:
```suggestion
Fixed invalid legacy permission fetching for comment mentions
```

File: api/src/services/comments.ts [hunk 3]
```diff
@@ -147,18 +148,23 @@ export class CommentsService extends ItemsService {
...
 			});
 
 			const accountability: Accountability = {
 				user: userID,
 				role: user['role']?.id ?? null,
-				admin: user['role']?.admin_access ?? null,
-				app: user['role']?.app_access ?? null,
+				admin: false,
+				app: false,
 				roles: await fetchRolesTree(user['role']?.id, this.knex),
```
Comment (ComfortablyCoding) [inline api/src/services/comments.ts:159]:
```suggestion
				roles: await fetchRolesTree(user['role']?.id ?? null, this.knex),
```

**Review by ComfortablyCoding — APPROVED** [review]
LGTM 🚀

**Pull request merged**